### DISCUSSION TITLE
Running container as non-root user?

### DISCUSSION THREAD

**Discussion by klaus1k** [alangrainger/immich-public-proxy#23] [Feature Requests]
Hello, is it possible to have the container run as a non-root user?
I tried adding `user: 1000.1000` to the compose file, but it threw a bunch of errors. it apparently can't create a `.pm2` folder - creating it manually for that user has no effect.
The mount point is owned by the same user, so that should not be the problem.

[immich-public-proxy-2024-11-16T11-20-31.log](https://github.com/user-attachments/files/17784458/immich-public-proxy-2024-11-16T11-20-31.log)

**Comment by alangrainger** [comment]
The container already runs as the non-root `node` user: 

https://github.com/alangrainger/immich-public-proxy/blob/main/Dockerfile#L17

I'm running this on rootless Podman myself.

Here's the running container, the UID is already 1000:

![Screenshot_20241116-124749](https://github.com/user-attachments/assets/9fd13eed-25cb-4935-8aac-aaaa3c3e1dfc)

**Reply by klaus1k** [reply]
Thanks for the quick reply!
I tried running the container as my user `1001` - for that I got the above errors, but since it is running as `1000` it's perfectly fine for me!
I just assumed that it is running as `root`,  as many other containers in their standard configuration.

**Reply by alangrainger** [reply]
No probs.

In the future just log into a container and type `id` or `whoami` to find details of the running user.

**Comment by alangrainger** [comment]
I'll close this since the container is already non-root, but feel free to reply with further details so we can figure out what's not working right for you. 

Please see my response above ^^^
